server/modules/data-access/cachedSingleAccountDynamoTable.js

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) Trainline Limited, 2016-2017. All rights reserved. See LICENSE.txt in the project root for license information. */
+
+'use strict';
+
+let mkArn = require('modules/data-access/dynamoTableArn').mkArn;
+let attachAuditMetadata = require('modules/data-access/dynamoAudit').attachAuditMetadata;
+let describeDynamoTable = require('modules/data-access/describeDynamoTable');
+let hashKeyAttributeName = require('modules/data-access/dynamoTableDescription').hashKeyAttributeName;
+let tableArn = require('modules/data-access/dynamoTableDescription').tableArn;
+let dynamoVersion = require('modules/data-access/dynamoVersion');
+let dynamoTableCache = require('modules/data-access/dynamoTableCache');
+let dynamoSoftDelete = require('modules/data-access/dynamoSoftDelete');
+let fp = require('lodash/fp');
+
+function constructor(physicalTableName, { ttl }) {
```

Comment:
Wary of naming this `constructor` as I'd assumed we'd then be creating a `new` instance somewhere. If this is just to be used as a closure would it be better named with `create` or `factory` or something similar?